mantidproject/mantid
# Add MSD fit parameter estimation to Indirect Data Analysis GUI

**Description of work.**
This PR introduces fit parameter estimations to the MSD fitting tab. In doing so it updates the following behavior:

- The first and last data points are no longer thrown away when constructing the guess workspace #28975 
- The data used to estimate the fit parameters updates with the current fitting range. 

This PR creates a fit parameter estimation class which may be used to easily add estimation to each tab within the GUI.

A couple of issues were found while testing this, which will be dealt with in another PR

- The graph is relimed on every plot, which means changing the fit range (to change the plot guess) will lead to the axis frequently changing. This will be fixed in a separate issue.
- When changing fit functions the previous fit isn't cleared. #28991

**To test:**
Code review
Functional test:
- Data to test each tab of the Indirect Data Analysis GUI can be found in #26631.
- Open the Indirect Data Analysis GUI
- Go to the MSD fitting tab
- Load Data

1. Add a guess workspace. Note that it will now create a guess workspace with the full range of data
2. The initial guess won't be great, as it finds the height from the first point in the range.
3. Move the fitting range to the peak of the data, the guess will update.



Fixes #28526  
Fixes #28975 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
